### PR TITLE
Create two test datasets to track low and high entropy

### LINKED ISSUE #46: Create two test datasets to track low and high entropy
### Existing issue check

- [X] I found no existing [issues](https://github.com/software-gardening/almanac/issues) covering this topic.

### Description

I want to add user-specified data to two separate markdown files, each located in their own  folders, so that I can maintain proper stability and tracking within the .git repository.

### Example

Using sub-processes to add/commit on the markdown files, allowing us to track changes in line content of each file directory.

### Would you like to work on a solution for this?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by willdavidson05**
# Description

This PR creates two separate markdown files representing low and high entropy. A function was created to add data to each file. For the data to be added and committed to the markdown files, subprocesses were used, initially adding a single line of code. Then, another commit adds six predictable lines to the low entropy file and 30 complex lines to the high entropy file.

This all done through referencing an add_entropy function, which allows for discretion of content being added to the test datasets.

The files are then zipped to ensure proper stabilization of the .git folders. This setup will provide test data that can be used as we dive deeper into the project.

I  appreciate any suggestions or feedback!

closes #46 

## What is the nature of your change?

- [X] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [X] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own contributions.
- [X] I have commented my content, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to related documentation (outside of book content).
- [X] My changes generate no new warnings.
- [X] New and existing tests pass locally with my changes.
- [X] I have added tests that prove my additions are effective or that my feature works.
- [X] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: tests/data/jupyter-book/high_entropy/high_entropy.md
Comment (d33bs) [inline tests/data/jupyter-book/high_entropy/high_entropy.md:1]: Seeing how this content repeats, consider practicing [don't repeat yourself (DRY)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) with this content by relying on the Python code or the Markdown files (but not both).

**Review by d33bs — COMMENTED** [review]
Nice job! I left a few comments and suggestions throughout this review. Please don't hesitate to let me know if you have any questions, happy to go over things with a conversation here or otherwise.

In addition, I wondered: what triggers or runs `tests/data/jupyter-book/add_data.py`? Would it make sense to spell this out with either code or comments?

**Comment by willdavidson05** [comment]
Thanks for the review and feedback @d33bs !! I'll start working on these comments now.

File: tests/data/almanack/entropy/add_data.py
Comment (falquaddoomi) [inline tests/data/almanack/entropy/add_data.py]: I'd suggest moving this code, as well as the calls to `zip_git_folder()` later in the file, to a "main" function and calling it from an `if __name__ == '__main__': main()` block, like you did in `add_entropy.py`.

Aside from being a convention, wrapping that code in a function that only executes if the script is directly run makes it possible to import the `zip_git_folder()` from another Python module without triggering all the side effects in this script (e.g., making git repos and zipping them).

**Review by falquaddoomi — COMMENTED** [review]
Nice work! Looks like @d33bs mentioned most of what I wanted to say, but I left a small comment.

File: tests/data/jupyter-book/add_data.py
Comment (willdavidson05) [reply tests/data/jupyter-book/add_data.py]: I have added docstrings for both files. Please let me know what you think, and if there is anything I should add or delete. Or if there is a better way to format it.

File: tests/data/jupyter-book/add_data.py
Comment (willdavidson05) [reply tests/data/jupyter-book/add_data.py]: Yes they would. In my edited code 
```python
    with zipfile.ZipFile(output_zip_path, "w") as zipf:
        for root, _, files in os.walk(folder_path):
            for file in files:
                file_path = os.path.join(root, file)
                zipf.write(file_path, os.path.relpath(file_path, folder_path))
``` 
we iterate through all files in the folder_path directory and adds them to the zip file. So as long as the .git directories exist within this directory, they will be included in the zip.

This is then called later 
```python
    zip_git_folder("high_entropy/.git", "high_entropy_git.zip")
    zip_git_folder("low_entropy/.git", "low_entropy_git.zip")
``` 
I added better titles and names since I can see what I had before did not provide enough context. Please let me know what you think!

File: tests/data/jupyter-book/add_data.py
Comment (willdavidson05) [reply tests/data/jupyter-book/add_data.py]: Would you be able to provide me with an example? I'm not sure how to go about this in a different way! Thanks!

File: tests/data/jupyter-book/high_entropy/high_entropy.md
Comment (willdavidson05) [reply tests/data/jupyter-book/high_entropy/high_entropy.md:1]: I removed this python code 
```python
        def entropy(data):
            from collections import Counter
            import math

            count = Counter(data)
            length = len(data)
            return -sum(
                frequency / length * math.log(frequency / length, 2)
                for frequency in count.values()
            )


        print(entropy("Random Data 1234!"))
``` 
from the add_entropy.py file subsequently removing it from the .md. Is this what you meant?

File: tests/data/jupyter-book/add_data.py
Comment (willdavidson05) [reply tests/data/jupyter-book/add_data.py]: Thank you for this idea, I made a few tweaks to the code. Instead of manually creating directories, I used pathlib.Path().parent.mkdir() to create the parent directories for the Markdown files. 
```python
pathlib.Path(md_file).parent.mkdir(parents=True, exist_ok=True)
``` 
Then, I wrote the baseline content to each Markdown file using a loop. After that, I ran the add_entropy.py script as before. Lastly, I ensured the .git folders were present using
```python
    (pathlib.Path("high_entropy") / ".git").mkdir(exist_ok=True)
    (pathlib.Path("low_entropy") / ".git").mkdir(exist_ok=True)
```

File: tests/data/almanack/entropy/add_data.py
Comment (willdavidson05) [reply tests/data/almanack/entropy/add_data.py]: Thanks for the suggestion! I've restructured the code as advised, wrapping the main functionality in a main() function and calling it from an if __name__ == '__main__': block.

**Comment by willdavidson05** [comment]
Thank you for the review @d33bs @falquaddoomi !! I have added changes and made comments to your feedback. I have requested another review, so please let me know if I missed anything or if you have any more ideas! Also no rush to get to it! Thank you!

File: tests/data/jupyter-book/add_data.py
Comment (d33bs) [reply tests/data/jupyter-book/add_data.py]: Looks great, cheers! Feel free to resolve.

File: tests/data/jupyter-book/add_data.py
Comment (d33bs) [reply tests/data/jupyter-book/add_data.py]: Nice approach!

File: tests/data/almanack/entropy/add_data.py
Comment (d33bs) [inline tests/data/almanack/entropy/add_data.py:16]: How should this script be invoked with context to the development environment? Consider documenting that somewhere in the docstring here.

File: tests/data/almanack/entropy/add_entropy.py
Comment (d33bs) [inline tests/data/almanack/entropy/add_entropy.py:61]: Consider relying on solely the execution of `main()` in `add_data.py` (which imports and uses `add_entropy`). Making this change would allow you to use `add_data.py` as a single entrypoint for the process instead of using multiple. Another way to think about this would be to consider how a developer would run these scripts, for ex. `poetry run python add_data.py`, etc.

```suggestion
```

**Review by d33bs — COMMENTED** [review]
Nice work! Thank you for responding to the comments and making updates! I left some additional comments. In addition I noticed that the zip files might be missing. Consider adding these so they may be retained with the source.

File: tests/data/almanack/entropy/add_data.py
Comment (d33bs) [inline tests/data/almanack/entropy/add_data.py:38]: Consider using `pathlib` for anywhere `os` is used here to help keep a consistent use of paths throughout.

File: tests/data/almanack/entropy/add_data.py
Comment (willdavidson05) [reply tests/data/almanack/entropy/add_data.py:19]: I didn't think of that, thanks Dave! This is what i added
"- To invoke this script within the development environment, use the following command:
  poetry run python add_data.py"
Let me know what you think!

Comment (d33bs) [reply tests/data/almanack/entropy/add_data.py:19]: Thanks, looks good!

File: tests/data/almanack/entropy/high_entropy_git.zip
Comment (d33bs) [inline tests/data/almanack/entropy/high_entropy_git.zip:1]: 🙌

**Review by d33bs — APPROVED** [review]
Cheers @willdavidson05, this LGTM! Feel free to merge when ready.

**Pull request merged**
